components/raftstore-v2/src/operation/command/mod.rs

```diff
@@ -306,6 +308,22 @@ impl<EK: KvEngine, ER: RaftEngine> Peer<EK, ER> {
             apply_res.applied_index,
             progress_to_be_updated,
         );
+        if self.pause_for_recovery()
+            && self.storage().entry_storage().commit_index() <= apply_res.applied_index
+        {
+            info!(self.logger, "recovery completed"; "apply_index" => apply_res.applied_index);
+            self.set_pause_for_recovery(false);
+            // Flush to avoid recover again and again.
+            if let Some(scheduler) = self.apply_scheduler() {
+                scheduler.send(ApplyTask::ManualFlush);
+            }
+            self.add_pending_tick(PeerTick::Raft);
+        }
+        if !self.pause_for_recovery() && self.storage_mut().apply_trace_mut().should_flush() {
```

Comment:
I don't think we should force flush here. It should be handled in rocksdb layer. 
What's more, 4K's index limit means the actual memtable needed is just 4MB for 1K value size (which is fairly large in normal cases).
Last time I tested 16MB memtable and actually it lowers the throughput than 32MB. For this change, I'm afraid it will further lower the throughput.

Reply:
As explained above, KVs are batched up instead proposing one by one. And the batch soft limit is 2MiB.